openshift_performance/ose3_perf/scripts/build_test_1.py

```diff
@@ -0,0 +1,414 @@
+    else:
+        selected_builds = all_builds[0:]
+
+    if global_config["shuffle"]:
+        selected_builds = random.sample(selected_builds,
+                                        len(selected_builds))
+
+    if global_config["batch"] == 0:
+        batch_size = len(selected_builds)
+    else:
+        batch_size = global_config["batch"]
+
+    while len(selected_builds) > 0:
+        this_batch_count = 0
+        futures = []
+        while this_batch_count < batch_size and len(selected_builds) > 0:
+            build = selected_builds.pop()
+            this_batch_count += 1
+            futures.append(executor.submit(run_build, build))
+
+        logger.info("All threads started, starting builds")
+        wait(futures)
+        time.sleep(int(global_config["sleep_time"]))
+
+
+def get_build_configs():
+    all_builds = []
+
+    build_configs = json.loads(run("oc get --all-namespaces=true -o json bc"))
+    if build_configs:
+        for build in build_configs["items"]:
+            all_builds.append({"namespace": build["metadata"]["namespace"],
+                               "name": build["metadata"]["name"]})
+    return all_builds
+
+
+def start():
+    global global_config
+    global global_build_stats
+    parser = OptionParser()
+    parser.add_option("-m", "--master", dest="master",
+                      help="url of the OpenShift master to login to")
+    parser.add_option("-u", "--user", dest="oseuser",
+                      help="OpenShift user for login")
+    parser.add_option("-p", "--password", dest="osepass",
+                      help="OpenShift password for login")
+    parser.add_option("-a", "--all", dest="allbuilds", action="store_true",
+                      help="Run all builds in all projects")
+    parser.add_option("-l", "--shuffle", dest="shuffle", action="store_true",
+                      help="Shuffle the order of the selected builds")
+    parser.add_option("-f", "--file", dest="buildfile",
+                      help="JSON file with builds to run")
+    parser.add_option("-n", "--numiter", dest="num", default=1,
+                      help="Number of iterations")
+    parser.add_option("-b", "--batch", dest="batch", default=0,
+                      help="Number of iterations")
+    parser.add_option("-s", "--sleep", dest="sleep", default=0,
+                      help="Seconds to sleep between iterations")
+    parser.add_option("-r", "--random", dest="random", default=0,
+                      help="Number of builds to select randomly "
+                           "on each iteration")
+    parser.add_option("-z", dest="nologin", action="store_true", default=False,
+                      help="Bypass oc login")
+    parser.add_option("-d", "--debug", dest="debug", action="store_true",
+                      help="Debug messages")
+    parser.add_option("-w", "--worker", dest="worker", default=50,
+                      help="Number of workers")
+
```

Comment:
Need to update the README with the new -w parameter

Reply:
Done